Userland/Libraries/LibWeb/HTML/HTMLElement.cpp

```diff
@@ -405,7 +408,79 @@ static void run_focusing_steps(DOM::Node* new_focus_target, DOM::Node* fallback_
     auto new_chain = focus_chain(new_focus_target);
 
     // 8. Run the focus update steps with old chain, new chain, and new focus target respectively.
-    run_focus_update_steps(old_chain, new_chain, *new_focus_target);
+    run_focus_update_steps(old_chain, new_chain, new_focus_target);
+}
+
+// https://html.spec.whatwg.org/multipage/interaction.html#unfocusing-steps
+static void run_unfocusing_steps(DOM::Node* old_focus_target)
+{
+    // NOTE: The unfocusing steps do not always result in the focus changing, even when applied to
+    // the currently focused area of a top-level browsing context. For example, if the currently
+    // focused area of a top-level browsing context is a viewport, then it will usually keep its
+    // focus regardless until another focusable area is explicitly focused with the focusing steps.
+
+    auto is_shadow_host = [](DOM::Node* node) {
+        return is<DOM::Element>(node) && static_cast<DOM::Element*>(node)->shadow_root() != nullptr;
+    };
+
+    // 1. If old focus target is a shadow host whose shadow root's delegates focus is true,
+    //    and old focus target's shadow root is a shadow-including inclusive ancestor of the currently focused area
+    //    of a top-level browsing context's DOM anchor, then set old focus target to that currently focused area of a top-level browsing context.
```

Comment:
The wrapping here is strange, please try to maintain equal line length

Reply:
good point, this place looked pretty awkward. I re-flowed the block comments in this function to 120 columns. (VsCode rewrap extension is cool :D)

Reply:
120 seems like a sensible target, gonna use that myself 👍